adobe-sign-services.module/adobe-sign-services.karaf/adobe-sign-services.karaf-core/src/main/java/com/aftia/adobe/karaf/adobesign/core/agreement/SimpleAgreementService.java

```diff
@@ -18,5 +19,13 @@ public String createAgreement(AdobeSignApiClient apiClient, AdobeSignAuthenticat
             throws AdobeSignServiceException {
         return Agreement.generateAgreement(apiClient, authentication, agreementInfo, xApiUser, xOnBehalfOfUser);
     }
-    
+
+    @Override
+    public String createAgreementView(AdobeSignApiClient apiClient, AdobeSignAuthentication authentication,
+            String agreementId, AgreementViewInfo agreementViewInfo, String xApiUser, String xOnBehalfOfUser)
+            throws AdobeSignServiceException {
+        return Agreement.generateAgreementView(apiClient, authentication, agreementId, agreementViewInfo, xApiUser,
```

Comment:
Maybe validate that some of the variables are not null or empty like the agreementId this will help prevent NPE.